contracts/src/token/psp35/extensions/batch.rs

```diff
@@ -63,7 +74,11 @@ pub trait PSP35BatchInternal {
     ) -> Result<(), PSP35Error>;
 }
 
-impl<T: PSP35Internal + InkStorage> PSP35BatchInternal for T {
+impl<B, T> PSP35BatchInternal for T
+where
+    B: BalancesManager + SpreadLayout + SpreadAllocate,
```

Comment:
```suggestion
    B: BalancesManager,
```

And check please other places)